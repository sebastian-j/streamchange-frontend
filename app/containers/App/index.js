/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';

import HomePage from 'containers/HomePage/Loadable';
import History from 'containers/History/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';

import GlobalStyle from '../../global-styles';

export default function App() {
  return (
    <div>
      <Switch>
        <Route exact path="/giveaway" component={HomePage} />
        <Route exact path="/giveaway-history" component={History} />
        <Route component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </div>
  );
}
