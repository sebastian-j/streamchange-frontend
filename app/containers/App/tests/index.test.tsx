import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import App from '../index';

const shallowRenderer = createRenderer();

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<App />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
