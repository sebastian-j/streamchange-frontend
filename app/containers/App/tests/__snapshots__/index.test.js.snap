// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/giveaway"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/giveaway-history"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/queue"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <GlobalStyleComponent />
</div>
`;
